### PR TITLE
Refactor Summary constructors

### PR DESCRIPTION
**Pull request by jokva**
Enable constructing Summary objects without an EclipseState, and add a
series of EclipseState overloads that properly instantiates a Summary
object from its members.

### CODE REVIEW

File: opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -712,6 +703,21 @@ Summary::Summary( const EclipseState& st,
     }
 }
 
+Summary::Summary( const EclipseState& es ) : Summary( es, es.getTitle() ) {}
+
+Summary::Summary( const EclipseState& es, const std::string& basename) :
+    Summary( es, basename.c_str() ) {}
+
+Summary::Summary( const EclipseState& es, const char* basename ) : Summary(
+        es.getSummaryConfig(),
```
Comment (joakim-hove) [inline opm/output/eclipse/Summary.cpp:712]: `es.getSummaryConfig():` "it works on my machine"?

**Comment by joakim-hove** [comment]
I guess a commit is missing in parser? Anyway - will merge when Travis is happy.

**Comment by jokva** [comment]
Yeah, a commit is missing. I'll re-start travis once all dependencies are sorted out.

**Comment by joakim-hove** [comment]
Is this alive?

**Comment by jokva** [comment]
It has conflicts, I can rebase it. It's a nice addition.

**Comment by jokva** [comment]
This PR is stale, but I suggest we keep it open until I've revised the changes and possibly made a new PR.

**Pull request closed without merging**